### PR TITLE
Contributing instructions

### LINKED ISSUE #1721: Contributing instructions
As a contributor, I'd like instructions so that I know how to do development and test my work.

### PR DESCRIPTION
**Pull request by dbeatty10**
Resolves #1721, unlocks #1341

## Overview

This is an initial set of instructions for doing development on this repository. It also includes a brief explanation how it relates to [Hubcap](https://github.com/dbt-labs/hubcap) and [`dbt deps`](https://docs.getdbt.com/reference/commands/deps) (from [dbt Core](https://github.com/dbt-labs/dbt-core)).

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+```bash
+bundle exec middleman serve --port 4567
+```
+
+### Use with `dbt deps` from dbt Core
+
+You can set the `DBT_PACKAGE_HUB_URL` environment variable to specify where `dbt deps` should look to resolve dependencies [listed in `packages.yml`](https://docs.getdbt.com/docs/building-a-dbt-project/package-management#how-do-i-add-a-package-to-my-project):
+```shell
+export DBT_PACKAGE_HUB_URL=http://localhost:4567/
```
Comment (joellabes) [inline CONTRIBUTING.md:48]: Perhaps instructions on how to delete this afterwards as well? Or at least a reminder _to_ delete it afterwards

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+```
+
+Or if you have a Netflify preview created by a pull request:
```
Comment (joellabes) [inline CONTRIBUTING.md:51]:
```suggestion
Or if you have a Netlify preview created by a pull request:
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+```
+
+Then `dbt deps` will use the JSON served from the following locations:
+- `data/packages/tnalpgge/cowsay/index.json`
+- `data/packages/tnalpgge/cowsay/versions/0.3.4.json`
+
+
+### Installing Ruby for Mac
+
+[https://antran.app/2021/m1_mac_part2/](https://antran.app/2021/m1_mac_part2/)
```
Comment (joellabes) [inline CONTRIBUTING.md:70]: What is this link for? Do I only need to read it if I have an M1 Mac? Does it replace the instructions below if I'm on an M1? Is it where the below instructions came from?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+```bash
+# Install rbenv
+# Follow the printed instructions to load rbenv automatically for your shell
+brew install rbenv
+
+# Initialise rbenv
+rbenv init
+
+# Verify rbenv
+curl -fsSL https://github.com/rbenv/rbenv-installer/raw/main/bin/rbenv-doctor | bash
```
Comment (joellabes) [inline CONTRIBUTING.md:81]: Meta note: a fun idea, but so grim that it's necessary. Packages are hell

**Review by joellabes — CHANGES_REQUESTED** [review]
A typo and a couple of things that could be clearer, but otherwise looking good!

File: CONTRIBUTING.md (hunk 2 above)
Comment (dbeatty10) [reply CONTRIBUTING.md]: Good catch -- thank you!

Who knowns, maybe "Netflify" is the next big thing in the tech world. Or at least the name of the next star pet on Instagram ("'Flify" for short).

File: CONTRIBUTING.md (hunk 4 above)
Comment (dbeatty10) [reply CONTRIBUTING.md:81]: 😂

File: CONTRIBUTING.md (hunk 1 above)
Comment (dbeatty10) [reply CONTRIBUTING.md:50]: Very good point! Committed an update to incorporate this feedback.

File: CONTRIBUTING.md (hunk 3 above)
Comment (dbeatty10) [reply CONTRIBUTING.md]: Committed an update to incorporate this feedback.

**Review by joellabes — APPROVED** [review]

**Pull request merged**